hedera-node/src/main/java/com/hedera/services/ServicesState.java

```diff
@@ -363,6 +371,35 @@ public RecordsRunningHashLeaf runningHashLeaf() {
 		return metadata.getUniqueTreasuryOwnershipAssociations();
 	}
 
+	private void updateLinks() {
+		var accounts = accounts();
+		var tokenRels = tokenAssociations();
+
+		for (var accountId : accounts.keySet()) {
+			var merkleAccount = accounts.getForModify(accountId);
+
+			EntityNumPair prevAssociationKey = MISSING_NUM_PAIR;
+			for (var tokenId : merkleAccount.tokens().asTokenIds()) {
+				var associationKey = EntityNumPair.fromLongs(accountId.longValue(), tokenId.getTokenNum());
+				var association = tokenRels.getForModify(associationKey);
+				association.setKey(associationKey);
+				association.setNextKey(prevAssociationKey);
+
+				if (prevAssociationKey != MISSING_NUM_PAIR) {
+					var prevAssociation = tokenRels.getForModify(prevAssociationKey);
+					prevAssociation.setPrevKey(associationKey);
+					tokenRels.put(prevAssociationKey, prevAssociation);
+				}
+
+				tokenRels.put(associationKey, association);
+				prevAssociationKey = associationKey;
+			}
+
+			merkleAccount.setLastAssociatedToken(prevAssociationKey);
```

Comment:
At this point we should be able to remove the `MerkleAccountTokens` child from this `merkleAccount`, right?

We might have a method in `MerkleAccount` like,

```
public void forgetAssociatedTokens() {
    addDeserializedChildren(List.of(state(), records()));
}
```

...and call `merkleAccount.forgetAssociatedTokens()` here.

Reply:
Nice.. thanks @tinker-michaelj